src/Umbraco.Cms.ManagementApi/Controllers/UpgradeController.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.ManagementApi.Filters;
+using Umbraco.Cms.ManagementApi.ViewModels.Installer;
+using Umbraco.New.Cms.Core.Factories;
+using Umbraco.New.Cms.Core.Models.Installer;
+using Umbraco.New.Cms.Core.Services.Installer;
+using Umbraco.New.Cms.Web.Common.Routing;
+
+namespace Umbraco.Cms.ManagementApi.Controllers;
+
+[ApiController]
+[ApiVersion("1.0")]
+[RequireRuntimeLevel(RuntimeLevel.Upgrade)]
+[BackOfficeRoute("api/v{version:apiVersion}/upgrade")]
+public class UpgradeController : Controller
```

Comment:
This controller needs authentication, but that's still in the works. I suggest leaving a TODO here, so we can catch up on the missing authentication once a new authentication scheme is in place.